platforms/android/library/src/main/java/io/element/android/wysiwyg/EditorStyledTextView.kt

```diff
@@ -135,37 +137,64 @@ open class EditorStyledTextView : AppCompatTextView {
         )
     }
 
+    // Tracks the initial touch coordinates of a touch event, they will be used to determine if a click event happened
+    private var initialTouchCoordinates: Pair<Float, Float>? = null
+    // Tracks whether a click event may be happening between touch event phases
+    private var maybeClickEvent = false
+
     override fun onTouchEvent(event: MotionEvent?): Boolean {
-        return when (event?.action) {
+        when (event?.action) {
             MotionEvent.ACTION_DOWN -> {
-                // Allow ACTION_UP to be called
+                // A press was detected, this may be a click
+                maybeClickEvent = true
+                // Call default touch handler to detect the start of normal taps and gestures
                 super.onTouchEvent(event)
-                true
+                // Store initial touch coordinates
+                initialTouchCoordinates = event.x to event.y
+                // Return that we handled the event, it'll allow us to receive `ACTION_UP` later
+                return true
             }
-            MotionEvent.ACTION_UP -> {
-                // Find selection matching the pointer coordinates
-                val offset = getOffsetForPosition(event.x, event.y)
-                // Look for clickable spans in that position
-                val spans = (text as? Spanned)?.getSpans(offset, offset, Any::class.java) ?: return false
-                val linkSpans = spans.filterIsInstance<LinkSpan>()
-                val pillSpans = spans.filterIsInstance<PillSpan>()
-                val customMentionSpans = spans.filterIsInstance<CustomMentionSpan>()
-                // Notify the link has been clicked
-                for (span in linkSpans) {
-                    if (span.url == null) continue
-                    onLinkClickedListener?.invoke(span.url)
+            MotionEvent.ACTION_MOVE -> {
+                val distance = initialTouchCoordinates?.let { hypot(it.first - event.x, it.second - event.y) } ?: 0f
+                // If the distance between the initial coordinates and the current ones is too big (4dp), we can assume it's not a click
```

Comment:
I am worried that manual detection of scroll / click gestures could be a bit error prone and lead to something which is either too sensitive or not sensitive enough. I wonder if there is a way to avoid implementing this ourselves manually?

I saw [`GestureDetector`](https://developer.android.com/reference/android/view/GestureDetector) which looks like it does something similar - is it worth looking into?

Reply:
`SimpleGestureDetector` does just this and simplifies the code **a lot**. Thanks for the suggestion!